src/compositions/useRoute.ts

```diff
@@ -1,13 +1,15 @@
-import { ExtractRouterRouteParamTypes, RouterRoute } from '@/types'
-import { Identity } from '@/types/utilities'
+import { useRouter } from '@/compositions/useRouter'
+import { UseRouteInvalidError } from '@/errors'
+import { RegisteredRouteMap, ResolvedRoute } from '@/types'
 
-type Route<T extends RouterRoute> = {
-  name: string,
-  params: ExtractRouterRouteParamTypes<T>,
-  query: unknown,
-  hash: string,
-}
+export function useRoute<TRouteName extends keyof RegisteredRouteMap>(routeName: TRouteName): ResolvedRoute<TRouteName>
+export function useRoute(routeName: string): ResolvedRoute {
```

Comment:
I think we should probably overload this and make `routeName` optional. If its not passed in the user just wants the route no matter what it is. Vs if they pass a name they want to to type guard.